Maui.RevenueCat.InAppBilling/Platforms/Android/RevenueCatBillingAndroid.cs

```diff
@@ -56,55 +56,55 @@ public partial class RevenueCatBilling : IRevenueCatBilling
 
     public async partial Task<List<OfferingDto>> LoadOfferings(bool forceRefresh, CancellationToken cancellationToken)
     {
-        if (!forceRefresh && !_cachedOfferingPackages.IsNullOrEmpty())
+        if (!forceRefresh && _cachedOfferingPackages != null)
         {
-            return _cachedOfferingPackages.ToOfferDtoList(); ;
+            return _cachedOfferingPackages.ToOfferingDtoList().ToList();
         }
 
         try
         {
-            using var offerings = await Purchases.SharedInstance.GetOfferingsAsync(cancellationToken);
-            if (offerings is null)
+            _cachedOfferingPackages = await Purchases.SharedInstance.GetOfferingsAsync(cancellationToken);
+            if (_cachedOfferingPackages is null)
             {
                 return new();
             }
 
-            using var currentOffering = offerings.Current;
-            if (currentOffering is null)
-            {
-                return new();
-            }
-
-            _cachedOfferingPackages = currentOffering.AvailablePackages.ToList();
-
-            return _cachedOfferingPackages.ToOfferDtoList(); ;
+            return _cachedOfferingPackages.ToOfferingDtoList().ToList();
         }
         catch (Exception ex)
         {
-            _logger.LogError(ex, $"{nameof(LoadOfferings)} didn't suceed.");
+            _logger.LogError(ex, $"{nameof(LoadOfferings)} didn't succeed.");
             return new();
         }
     }
-    public async partial Task<PurchaseResult> PurchaseProduct(string offeringIdentifier, CancellationToken cancellationToken)
+    public async partial Task<PurchaseResult> PurchaseProduct(string offeringIdentifier, string packageIdentifier, CancellationToken cancellationToken)
     {
         if (!_isInitialized)
         {
             _logger.LogError($"To call {nameof(PurchaseProduct)} you firstly have to call Initialize method.");
             throw new Exception("RevenueCatBilling wasn't initialized");
         }
 
-        var packageToBuy = _cachedOfferingPackages.FirstOrDefault(p => p.Identifier == offeringIdentifier);
-        if (packageToBuy is null)
+        if (_currentActivityContext is null)
         {
-            _logger.LogError("No offering/packege with identifier: {offeringIdentifier} found. Make sure you called LoadOfferings before.", offeringIdentifier);
-            throw new Exception($"No offering/packege with identifier: {offeringIdentifier} found. Make sure you called LoadOfferings before.");
+            throw new Exception("Android Current Activity can't be null.");
         }
 
-        if (_currentActivityContext is null)
+        var offeringToBuy = _cachedOfferingPackages.GetOffering(offeringIdentifier);
+        if (offeringToBuy is null)
         {
-            throw new Exception("Android Current Activity can't be null.");
```

Comment:
after you made Offerings explicitly nullable it shows a warning `may be null here` on L93. Same on iOS